libs/screen/sim/image.ts

```diff
@@ -933,6 +933,412 @@ namespace pxsim.ImageMethods {
         }
         return false;
     }
+
+    const TWO_PI = 2 * Math.PI;
+    const HALF_PI = Math.PI / 2;
+    const THREE_HALF_PI = 3 * Math.PI / 2
+
+    const FX_ONE = 1;
+
+    function fxMul(a: number, b: number) {
+        return (a * b)
+    }
+
+    function fxDiv(a: number, b: number) {
+        return a / b;
+    }
+
+    function fxToInt(v: number) {
+        return v
+    }
+
+    function fxFloor(v: number) {
+        return v | 0;
+    }
+
+    interface ParsedShearArgs {
+        sx: number;
+        sy: number;
+        scaledWidth: number;
+        scaledHeight: number;
+        minX: number;
+        minY: number;
+        maxX: number;
+        maxY: number;
+        xShear: number;
+        yShear: number;
+        flip: boolean;
+    }
+
+    function parseShearArgs(src: RefImage, args: RefCollection, argIndex: number): ParsedShearArgs {
+        const parsed: ParsedShearArgs = {
+            sx: 0,
+            sy: 0,
+            scaledWidth: 0,
+            scaledHeight: 0,
+            minX: 0,
+            minY: 0,
+            maxX: 0,
+            maxY: 0,
+            xShear: 0,
+            yShear: 0,
+            flip: false
+        };
+
+        const sx = ((args.getAt(argIndex) as number) * FX_ONE);
+        const sy = ((args.getAt(argIndex + 1) as number) * FX_ONE);
+        let angle = args.getAt(argIndex + 2) as number;
+
+        parsed.sx = sx;
+        parsed.sy = sy;
+
+        if (sx <= 0 || sy <= 0) {
+            return parsed;
+        }
+
+        angle %= TWO_PI;
+        if (angle < 0) {
+            angle += Math.PI;
+        }
+
+        let flip = false;
+        if (angle > HALF_PI && angle <= THREE_HALF_PI) {
+            flip = true
+            angle = (angle + Math.PI) % TWO_PI
+        }
+
+        const xShear = (-Math.tan(angle / 2) * FX_ONE);
+        const yShear = (Math.sin(angle) * FX_ONE);
+
+        const scaledWidth = src._width * sx;
```

Comment:
It seems the TS implementation isn't actually using fixed point, but if it were, this line would be mixing fixed and floating point math.

Reply:
this is fine, though, since we're dealing with integers and this would just require unnecessary bit shifting to convert into FX and immediately convert back